docs/HyperIndex/Guides/rpc-requests.md

```diff
@@ -0,0 +1,324 @@
+---
+
+# RPC Requests guide
+
+TLDR; The repo for the code base can be found [here](todo_add_link)
+
+**Scenario**  
+In this guide, we aim to get token information for every token involved in a [Uniswap V3 pool creation event](https://docs.uniswap.org/contracts/v3/reference/core/interfaces/IUniswapV3Factory#poolcreated). For each token we should index its name, symbol and decimals.
+
+**Problem**  
+At first glance this may appear to be a simple task as we are only indexing over one event, the [pool created](https://docs.uniswap.org/contracts/v3/reference/core/interfaces/IUniswapV3Factory#poolcreated) event. However, there is one catch, have a look at the event signature: 
+
+```yaml
+PoolCreated(address indexed token0, address indexed token1, uint24 indexed fee, int24 tickSpacing, address pool)
+```
+
+The event only provides the token addresses, not the token name, symbol, and decimals. So how are we to fetch this extra information in our event handler? 
+
+**Solution**  
+In order to do this, we need to make RPC requests to the token contract.
+
+### Prerequisites
+
+This guide assumes basic familiarity with the viem library for making contract calls. Check out the [viem documentation](https://viem.sh/) for more information. This [medium article](https://medium.com/@0xape/typescript-and-viem-quickstart-for-blockchain-scripting-3f1846970b6f) is highly recommended for a gentle introduction to viem with a very similar example to what we are doing here.
+
+### Part 1: Create our Uniswap V3 `poolcreated` indexer
+
+`npx envio init`
+Contract address: `0x1F98431c8aD98523631AE4a59f267346ea31F984`
+
+We then make some light modifications to remove unnecessary events and simplify the schema. The resulting config, schema, and event handlers look as follows:
+
+> config.yaml
+
+:::info
+We remove the `FeeAmountEnabled` and `OwnerChanged` events as they are not relevant to our use case.
+:::
+
+```yaml
+name: uniswap-v3-factory-token-indexer
+networks:
+- id: 1
+  start_block: 12369620
+  contracts:
+  - name: UniswapV3Factory
+    address:
+    - 0x1F98431c8aD98523631AE4a59f267346ea31F984
+    handler: src/EventHandlers.ts
+    events:
+    - event: PoolCreated(address indexed token0, address indexed token1, uint24 indexed fee, int24 tickSpacing, address pool)
+rollback_on_reorg: false
+```
+
+> shema.graphql
+
+:::info
+Simplify our schema to only include the pool and token information.
+:::
+
+```graphql
+type Token {
+  id: ID! #address
+  name: String!
+  symbol: String!
+  decimals: Int!
+}
+
+type Pool {
+  id: ID! # address
+  token0: Token! 
+  token1: Token!
+  fee: BigInt!
+  tickSpacing: BigInt!
+  pool: String!
+}
+```
+
+> src/EventHandlers.ts
+
+```typescript
+import {
+  UniswapV3Factory,
+  Pool,
+} from "generated";
+import { getTokenDetails } from "./tokenDetails";
+
+UniswapV3Factory.PoolCreated.handler(async ({ event, context }) => {
+  const entity: Pool = {
+    id: `${event.chainId}_${event.block.number}_${event.logIndex}`,
```

Comment:
I usually use dashes instead of underscores, both are perfectly fine just might be confusing for some users so I think in general we should try stick to one. I couldn't find anything now in the docs with one or the other but I know all the examples I've done would have '-'
```suggestion
    id: `${event.chainId}-${event.block.number}-${event.logIndex}`,
```